### ISSUE TITLE
FragmentManager / Databinding Issue

### ISSUE THREAD

**Issue by virgil85** [mikepenz/AboutLibraries#504]
## About this issue

When adding classpath("com.mikepenz.aboutlibraries.plugin:aboutlibraries-plugin:8.1.6") to my build.gradle (root level) dependencies and applying the plugin in the app level - the application will build just fine but during runtime it does crash with an NullPointerException. 

Executing  ./gradlew exportLibraries works fine. I did not try adding the bundled activity.

```
2020-06-03 15:14:30.415 1931-1931/com.continental.android.intersectcollisionwarning E/AndroidRuntime:     at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManagerImpl.java:2100)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1874)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1830)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1727)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2663)
        at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2613)
        at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:2624)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:904)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2659)
        at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2613)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:246)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:542)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
        at com.continental.android.intersectcollisionwarning.base.views.BaseActivity.onStart(BaseActivity.kt:90)
        at com.continental.android.intercollisionwarning.home.ManagerActivity.onStart(ManagerActivity.kt:230)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1440)
        at android.app.Activity.performStart(Activity.java:8089)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3784)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:235)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:215)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:187)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:105)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2373)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:213)
        at android.app.ActivityThread.main(ActivityThread.java:8147)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1101)
     Caused by: java.lang.NullPointerException: entry
        at java.util.zip.ZipFile.getInputStream(ZipFile.java:395)
        at com.google.android.apps.gmm.map.util.jni.NativeHelper.unzip(NativeHelper.java:61)
        at com.google.android.apps.gmm.map.util.jni.NativeHelper.safeLoadLibrary(NativeHelper.java:48)
        	... 72 more
```

In my project data binding is enabled and the application does crash when inflating the view with a layout that does visualize a Google map. It won't crash here if the AboutLibraries plugin is not listed in the build.gradle. 

```
binding = DataBindingUtil.inflate(
            inflater ?: LayoutInflater.from(context),
            R.layout.fragment_dashboard_map,
            container,
            false
        )
```

## Details
- [8.1.6] Used library version
- [5.4.6] Used gradle build tools version
- [3.6.3] Used tooling / Android Studio version
- JDK 8
- Kotlin plugin 1.3.71
- AndroidX
- SDK Tools 26.1.1
- SDK Platform-Tools 30.0.0
- Android 10 / API Level 29

Am I missing something? Is anything necessary in AndroidManifest?

**Comment by mikepenz** [comment]
@virgil85 could it be that you exceed the multidex limit? 

if you do not apply the plugin but only use the dependency will it work?

Do you potentially have different versions of major libs like material components, or similar?

**Comment by virgil85** [comment]
@mikepenz since I have
`multiDexEnabled = true` 
I guess the multidex limit should not affect me. 

If I don't apply the plugin but only use the dependency I get the same runtime errors as before.

Yes, I'm using a lot of major libs (material components in version 1.1.0-beta01). Is there an easy way to list them all?

**Comment by mikepenz** [comment]
./gradlew app:dependencies gives you all dependencies and their version. 

It looks like the plugin is not the cause. And if it is only adding the library that indicates:

- multidex issue
- conflict of library versions

:)

**Comment by virgil85** [comment]
I just updated all library versions that I use to the same version as they are used in AboutLibraries.

So there should be no library conflicts regarding those libraries:
`androidX
appcompat
constraintLayout
kotlin
coreKtx
material
`

It will still crash during runtime while inflating a layout (in my case a fragment with activated data binding).

I understand that this is probably not a bug of the library but I may not be the only one with this issue - so maybe solving this compatibility issue might help others as well.

I will update the ticket when I find the reason/solution why this is happening or how I solved it.

**Comment by mikepenz** [comment]
the underlaying exception is here:

```
Caused by: java.lang.NullPointerException: entry
        at java.util.zip.ZipFile.getInputStream(ZipFile.java:395)
        at com.google.android.apps.gmm.map.util.jni.NativeHelper.unzip(NativeHelper.java:61)
        at com.google.android.apps.gmm.map.util.jni.NativeHelper.safeLoadLibrary(NativeHelper.java:48)
``` 

not sure what gmm tires to do here :D

**Comment by virgil85** [comment]
Ok - I found the issue(s):

First I removed

`packagingOptions {
        exclude("lib/arm64-v8a/libgmm-jni.so")
 }`

which I had because when I wrote the app the first time it was a quick fix.

After solving that I got the error, that some inline code was compiled jvm6 and not jvm8 so I added this to all the modules that threw the error:

`compileOptions {
        sourceCompatibility = JavaVersion.VERSION_1_8
        targetCompatibility = JavaVersion.VERSION_1_8
    }
kotlinOptions {
        jvmTarget = JavaVersion.VERSION_1_8.toString()
}`

At that moment the app started but when I tried to use the UI module I got the error that my theme has to be a descendent of MaterialComponents (it was AppCompat). So I changed to the Material Bridge Designs (which are some kind of extended AppCompat choices where I don't have to change my whole layout parameters):

`
    <style name="AppTheme" parent="Theme.MaterialComponents.Light.NoActionBar.Bridge">`

And now everything works.

Thank you for pointing me in the right direction.
